services/src/main/java/org/fao/geonet/api/reports/ReportUtils.java

```diff
@@ -137,25 +139,41 @@ public static String retrieveMetadataTitle(final String metadataUuid) {
      * <p>
      * Duplicated with XslUtil - to refactor.
      * <p>
-     * TODO / TODOES
+     * TODO / TODOES improve the management of the different type of fields
      *
      * @param metadataUuid Metadata identifier.
      * @param fieldName  Lucene field name.
+     * @param fieldNameKey  Key of the field name (eg 'default').
      * @return Field value.
      */
     private static String retrieveMetadataIndexField(
         final String metadataUuid,
-        final String fieldName) {
+        final String fieldName,
+        final String fieldNameKey) {
         EsSearchManager searchManager = ApplicationContextHolder.get().getBean(EsSearchManager.class);
         try {
             Map<String, Object> mdIndexFields = searchManager.getDocument(metadataUuid);
-            if ("title".equals(fieldName)) {
-                Object titleObjectField = mdIndexFields.get(IndexFields.RESOURCE_TITLE + "Object");
-                Object titleField = mdIndexFields.get(IndexFields.RESOURCE_TITLE);
-                if (titleObjectField instanceof HashMap) {
-                    return (String) ((HashMap<?, ?>) titleObjectField).get("default");
-                } else if (titleField instanceof String) {
-                    return (String) titleField;
+            Object field = mdIndexFields.get(fieldName);
+            System.out.println(mdIndexFields);
+            if (field instanceof HashMap) {
+                if (fieldNameKey != "") {
+                    return (String) ((HashMap<?, ?>) field).get(fieldNameKey);
+                } else {
+                    return (String) field.toString();
+                }
+            } else if (field instanceof String) {
+                return (String) field;
+            } else if (field instanceof ArrayList) {
+                Object fieldItem = ((ArrayList<?>) field).get(0);
+                if (fieldItem instanceof HashMap) {
+                    if (fieldNameKey != "") {
+                        return (String) ((HashMap<?, ?>) fieldItem).get(fieldNameKey);
+                    } else {
+                        return (String) fieldItem.toString();
+                    }
+                }
+                else if (fieldItem instanceof String) {
+                    return (String) field.toString().replace("[", "").replace("]", " ");
```

Comment:
```suggestion
                    return (String) fieldItem;
```
should be good enough ?